atc/engine/check_delegate.go

```diff
@@ -77,7 +77,7 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 
 	// rate limit periodic resource checks so worker load (plus load on external
 	// services) isn't too spiky
-	if !d.build.IsManuallyTriggered() && d.plan.Resource != "" {
+	if !d.build.IsManuallyTriggered() && (d.plan.Resource != "" || d.plan.ResourceType != "") {
```

Comment:
i think this is fine but i'm a little surprised to see this here - did something require it?

Reply:
Resource check will cause resource type check, so I consider resource type check should be handled in the same logic as resource check.